### PR TITLE
[Minor] Add image-id into implementation features

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/resources.py [hunk 1]
```diff
@@ -799,15 +800,17 @@ def _try_validate_image_id(self) -> None:
                     'Cloud must be specified when image_id is provided.')
 
         # Apr, 2023 by Hysun(hysun.he@oracle.com): Added support for OCI
```
Comment (romilbhardwaj) [inline sky/resources.py:802]: This line can probably be removed now : )

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll!

File: sky/resources.py [hunk 2]
```diff
@@ -799,15 +800,17 @@ def _try_validate_image_id(self) -> None:
...
+        try:
+            self._cloud.check_features_are_supported(
+                self,
+                requested_features={
+                    clouds.CloudImplementationFeatures.IMAGE_ID
+                })
+        except exceptions.NotSupportedError as e:
             with ux_utils.print_exception_no_traceback():
                 raise ValueError(
                     'image_id is only supported for AWS/GCP/IBM/OCI, please '
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/resources.py]: Thanks! Fixed.

File: sky/resources.py (hunk 1 above)
Comment (Michaelvll) [reply sky/resources.py]: Good point! Removed. Thanks!

**Pull request merged**